integtest.sh

```diff
@@ -102,17 +102,84 @@ echo -e "Test Files List:"
 echo $TEST_FILES | tr ',' '\n'
 echo "BROWSER_PATH: $BROWSER_PATH"
 
+# Array to store remote cypress workflow background processes IDs when run in parallel
+declare -a all_process_pids
+
+run_remote_cypress() {
+    local repo="$1"
+    local workflow_name="$2"
+    local os_url="$3"
+    local osd_url="$4"
+    local branch_ref="$5"
+
+    # Call the remoteCypress.sh script with the required arguments
+    source remoteCypress.sh -r "$repo" -w "$workflow_name" -o "$os_url" -d "$osd_url" -b "$branch_ref" &
+    bg_process_pid=$!
+    echo "PID for the repo $repo is : $bg_process_pid"
+    all_process_pids+=($bg_process_pid)
```

Comment:
i think it might be worth to consider using pid files https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/config/opensearch_dashboards.yml#L122

i didn't realize this logic existed but has long as the file is created we can reference the process id file instead of it in memory